test/test_filters.rb

```diff
@@ -594,6 +603,18 @@ def to_liquid
         assert_equal "b", results[1]["id"]
         assert_equal "d", results[2]["id"]
       end
+
+      should "filter posts" do
+        site = fixture_site.tap(&:read)
+        results = @filter.where_exp(site.posts, "obj", "obj.title == 'Foo Bar'")
```

Comment:
This should be `site.site_payload["site"]["posts"]` to mimic the Liquid context.